airbyte-webapp/src/components/EntityTable/components/StreamStatusCell.tsx

```diff
@@ -0,0 +1,178 @@
+  if (stream.config && stream.config.selected) {
+    if (stream.config.status === "active" && stream.config.latestSyncJobStatus !== "failed") {
+      if (
+        // This can be undefined due to historical data, but should always be present
+        stream.config.scheduleType &&
+        !["cron", "manual"].includes(stream.config.scheduleType) &&
+        stream.config.latestSyncJobCreatedAt &&
+        stream.config.scheduleData?.basicSchedule?.units &&
+        stream.config.latestSyncJobCreatedAt * 1000 < // x1000 for a JS datetime
+          dayjs()
+            // Subtract 2x the scheduled interval and compare it to last sync time
+            .subtract(stream.config.scheduleData.basicSchedule.units, stream.config.scheduleData.basicSchedule.timeUnit)
+            .valueOf()
+      ) {
+        return "Behind";
+      }
+      return "On Track";
+    } else if (stream.config.latestSyncJobStatus === "failed") {
+      return "Error";
+    }
+  }
+  return "Disabled";
+};
+
+const sortStreams = (streams: AirbyteStreamWithStatusAndConfiguration[]): Record<StatusType, number> =>
+  streams.reduce(
+    (sortedStreams, stream) => {
+      sortedStreams[getStatusForStream(stream)]++;
+      return sortedStreams;
+    },
+    // This is the intended display order thanks to Javascript object insertion order!
+    { /* "Action Required": 0, */ Error: 0, Behind: 0, "On Track": 0, Disabled: 0 }
+  );
+
+const StreamsBar: React.FC<{ streams: AirbyteStreamWithStatusAndConfiguration[] }> = ({ streams }) => {
+  const sortedStreams = sortStreams(streams);
+  return (
+    <div className={styles.bar}>
+      {(Object.entries(sortedStreams) as Array<[StatusType, number]>)
+        .filter(([, count]) => !!count)
+        .map(([statusType, count]) => (
+          <div
+            style={{ width: `${Number(count / streams.length) * 100}%` }}
+            className={classNames(styles.filling, statusMap[statusType])}
+            key={statusType}
+          />
+        ))}
+    </div>
+  );
+};
+
+const StreamsPerStatus: React.FC<{ streams: AirbyteStreamWithStatusAndConfiguration[] }> = ({ streams }) => {
+  const sortedStreams = sortStreams(streams);
+  return (
+    <>
+      {(Object.entries(sortedStreams) as Array<[StatusType, number]>)
+        .filter(([, count]) => !!count)
+        .map(([statusType, count], index) => (
+          <div className={styles.tooltipContent} key={statusType}>
+            <div className={styles.streamsDetail}>
+              {iconMap[statusType]} <b>{count}</b> {statusType}
+            </div>
+            {streams[index].config?.isSyncing ? (
+              <div className={styles.syncContainer}>
+                {count} <Syncing className={styles.syncing} />
+              </div>
+            ) : null}
+          </div>
+        ))}
+    </>
+  );
+};
+
+const StreamStatusCellTooltipContent = ({ streams }: { streams: AirbyteStreamWithStatusAndConfiguration[] }) => {
```

Comment:
Semantics... but even though it's using a tooltip to render the content, this is more of a Popover component since the function of the tooltip is to explain something. `StreamStatusPopover`?

Reply:
sure!